bazel/cc_configure.bzl

```diff
@@ -1,56 +1,48 @@
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-"""Rules for configuring the C++ toolchain (experimental)."""
+def _quiet_fake_which(program):
+  return struct(_envoy_fake_which = program)
 
+def _noisy_fake_which(program):
+  return None
 
-load("@bazel_tools//tools/cpp:windows_cc_configure.bzl", "configure_windows_toolchain")
-load("@bazel_tools//tools/cpp:osx_cc_configure.bzl", "configure_osx_toolchain")
-load("//bazel:unix_cc_configure.bzl", "configure_unix_toolchain")
-load("@bazel_tools//tools/cpp:lib_cc_configure.bzl", "get_cpu_value")
+def _build_envoy_cc_wrapper(repository_ctx):
+  real_cc = find_cc(repository_ctx, {})
+  environ_cxx = repository_ctx.os.environ.get("CXX", "g++")
+  fake_os = struct(
+    environ = {"CC": environ_cxx},
+  )
+  real_cxx = find_cc(struct(
+      which = _quiet_fake_which,
+      os = fake_os,
+  ), {})
+  if hasattr(real_cxx, "_envoy_fake_which"):
+    real_cxx = repository_ctx.which(real_cxx._envoy_fake_which)
+    if real_cxx == None:
+      find_cc(struct(
+        which = _noisy_fake_which,
+        os = fake_os,
+      ), {})
+  repository_ctx.template("envoy_cc_wrapper", repository_ctx.attr._envoy_cc_wrapper, {
+      "{ENVOY_REAL_CC}": repr(str(real_cc)),
+      "{ENVOY_REAL_CXX}": repr(str(real_cxx)),
+  })
+  return repository_ctx.path("envoy_cc_wrapper")
 
-def cc_autoconf_impl(repository_ctx, overriden_tools = dict()):
-  repository_ctx.symlink(
-      Label("@bazel_tools//tools/cpp:dummy_toolchain.bzl"), "dummy_toolchain.bzl")
-  env = repository_ctx.os.environ
+def cc_autoconf_impl(repository_ctx):
   cpu_value = get_cpu_value(repository_ctx)
-  if cpu_value == "freebsd":
-    # This is defaulting to the static crosstool, we should eventually
-    # autoconfigure this platform too.  Theorically, FreeBSD should be
-    # straightforward to add but we cannot run it in a docker container so
-    # skipping until we have proper tests for FreeBSD.
-    repository_ctx.symlink(Label("@bazel_tools//tools/cpp:CROSSTOOL"), "CROSSTOOL")
-    repository_ctx.symlink(Label("@bazel_tools//tools/cpp:BUILD.static"), "BUILD")
-  elif cpu_value == "x64_windows":
-    # TODO(ibiryukov): overriden_tools are only supported in configure_unix_toolchain.
-    # We might want to add that to Windows too(at least for msys toolchain).
-    configure_windows_toolchain(repository_ctx)
-  elif (cpu_value == "darwin" and
-      ("BAZEL_USE_CPP_ONLY_TOOLCHAIN" not in env or env["BAZEL_USE_CPP_ONLY_TOOLCHAIN"] != "1")):
-    configure_osx_toolchain(repository_ctx, overriden_tools)
-  else:
-    configure_unix_toolchain(repository_ctx, cpu_value, overriden_tools)
+  overriden_tools = {}
+  if cpu_value not in ["freebsd", "x64_windows", "darwin"]:
+    overriden_tools["gcc"] = _build_envoy_cc_wrapper(repository_ctx)
```

Comment:
Why is gcc hardcoded here? Is this generic $CC effectively? If so, why not on Darwin as well?

Reply:
Added a comment. To Bazel, all CCs are GCC.